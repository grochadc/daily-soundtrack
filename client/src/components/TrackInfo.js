import React from "react";
import { Media, Panel } from "react-bootstrap";

export default function TrackInfo(props) {
  let { name, artists, album } = props;
  let title = name;
  let artist = artists[0].name;
  let art = album.images[1].url;

  return (
    <Panel>
      <Media>
        <Media.Left>
          <img alt={"Album Art"} src={art} height="100" width="100" />
        </Media.Left>
        <Media.Body>
          <p>
            <strong>Title:</strong> {title}
          </p>
          <p>
            <strong>Artist:</strong> {artist}
          </p>
          <p>
            <strong>Album:</strong> {album.name}
          </p>
        </Media.Body>
      </Media>
    </Panel>
  );
}
