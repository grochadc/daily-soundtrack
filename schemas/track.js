const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const trackSchema = new Schema({
  track_info: {
    title: String,
    artist: String,
    album: String,
    art: String
  },
  uri: String,
  date: Date,
  user: {
    id: String,
    name: String
  },
  message: String
});

module.exports = trackSchema;
