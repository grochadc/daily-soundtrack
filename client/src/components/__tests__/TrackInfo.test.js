import React, { Component } from "react";
import renderer from "react-test-renderer";
import track from "./track.json";
import TrackInfo from "../TrackInfo";

jest.mock("react-bootstrap");

test("Renders a track", () => {
  const component = renderer.create(<TrackInfo track={track} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
