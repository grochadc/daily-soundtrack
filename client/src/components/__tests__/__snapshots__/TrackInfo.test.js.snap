// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a track 1`] = `null`;
